core-dev/packages/coq/coq.dev/opam

```diff
@@ -1,54 +1,41 @@
+# This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
-maintainer: "coqdev@inria.fr"
-authors: "The Coq development team, INRIA, CNRS, and contributors."
-homepage: "https://coq.inria.fr/"
-bug-reports: "https://github.com/coq/coq/issues"
-dev-repo: "git+https://github.com/coq/coq.git"
-license: "LGPL-2.1-only"
-synopsis: "Formal proof management system"
+version: "dev"
+synopsis: "The Coq Proof Assistant"
 description: """
-The Coq proof assistant provides a formal language to write
-mathematical definitions, executable algorithms, and theorems, together
-with an environment for semi-interactive development of machine-checked
-proofs. Typical applications include the certification of properties of programming
-languages (e.g., the CompCert compiler certification project and the
-Bedrock verified low-level programming library), the formalization of
-mathematics (e.g., the full formalization of the Feit-Thompson theorem
-and homotopy type theory) and teaching.
-"""
+Coq is a formal proof management system. It provides
+a formal language to write mathematical definitions, executable
+algorithms and theorems together with an environment for
+semi-interactive development of machine-checked proofs.
 
-depopts: [
-  "coq-native"
-]
+Typical applications include the certification of properties of
+programming languages (e.g. the CompCert compiler certification
+project, or the Bedrock verified low-level programming library), the
+formalization of mathematics (e.g. the full formalization of the
+Feit-Thompson theorem or homotopy type theory) and teaching."""
+maintainer: ["The Coq development team <coqdev@inria.fr>"]
+authors: ["The Coq development team, INRIA, CNRS, and contributors"]
+license: "LGPL-2.1-only"
+homepage: "https://coq.inria.fr/"
+doc: "https://coq.github.io/doc/"
+bug-reports: "https://github.com/coq/coq/issues"
 depends: [
-  "ocaml" {>= "4.09.0"}
-  "ocamlfind" {build}
-  "dune" {>= "2.5.1"}
-  "conf-findutils" {build}
-  "zarith" {>= "1.10"}
-]
-conflicts: [
-  "base-nnp"
-  "ocaml-option-nnpchecker"
+  "dune" {>= "2.9"}
+  "coq-core" {= version}
+  "coq-stdlib" {= version}
 ]
 build: [
+  ["dune" "subst"] {pinned}
   [
-    "./configure"
-    "-configdir" "%{lib}%/coq/config"
-    "-prefix" prefix
-    "-mandir" man
-    "-docdir" "%{doc}%/coq"
-    "-libdir" "%{lib}%/coq"
-    "-datadir" "%{share}%/coq"
-    "-coqide" "no"
-    "-native-compiler" "yes" {coq-native:installed} "no" {!coq-native:installed}
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
   ]
-  [ make "COQ_USE_DUNE=" "-j%{jobs}%" ]
 ]
-install: [
-  [ make "COQ_USE_DUNE=" "install" ]
-]
-
-url {
-  src: "git+https://github.com/coq/coq.git#master"
-}
+dev-repo: "git+https://github.com/coq/coq.git"
```

Comment:
This looks fishy, and the same holds for the other packages. IMO this opam file is fine for `opam install https://github.com/coq/coq` but is not good for an opam repository

Reply:
Oh yes, sorry I am not familiar with how this works, I'll amend. Does this mean that we can't share the same opam files for stable opam releases, coq master and coq-core.dev ?

Reply:
> Does this mean that we can't share the same opam files for stable opam releases, coq master and coq-core.dev?

I do not understand. Why would you even want to share them? The build scripts, the required versions of dependencies, the conflicts, the patches, etc, everything at some point is bound to diverge.

Reply:
Why does it have to diverge, since 2.0, opam will update the build definitions from a pin.

For example, any divergence of coq.dev with what we have in master is just a bug. For stable branches, also the canonical files should be the ones in the repos.

What would be an example of such a diverenge? (Other than a patch not yet upstreamed)

Maybe the opam model is just a bit weird when compared to other systems.

Reply:
I agree, they should be in sync. Having the definition of the build process outside the repo is quite annoying, but at least it should be the same as what it is inside the repo.

> Does this mean that we can't share the same opam files for stable opam releases, coq master and coq-core.dev ?

Does opam complain if you just have the `url` thing in master?
What I do for Iris is to auto-generate the opam repo file from the git file by appending a suitable `url` block.

Reply:
Anyways this is fixed now, I'm gonna merge, sorry for the delay folks.

Reply:
This is the `coq.dev` package. There is nothing "stable" about it. It builds the latest git master commit. So it needs to be in sync with the build instructions for the latest git master commit.

Reply:
For stable branches, I think what they meant is that at the time the release is made, you want to have the build instructions in sync. Later you might have to fix them, and then they start to diverge, sure.

Reply:
In fact, more evidence for this point of view that as of today, opam repo maintainers will request to apply local fixes to the repos with a _must_.